src/errorBoundary.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+
+interface Properties {
+    message: React.ReactNode;
+}
+
+interface State {
+    hasError: boolean;
+}
+
+export class ErrorBoundary extends React.Component<Properties, State> {
+    constructor(props: Properties) {
+        super(props);
+        this.state = { hasError: false };

```

Comment:
can ErrorBoundary use hooks as well for its states?

Reply:
no, React hasn't provided a hook version for componentDidCatch/getDerivedErrorFromProps